src/FSharp.Data.GraphQL.Server.AspNetCore/Giraffe/HttpHandlers.fs

```diff
@@ -1,155 +1,268 @@
-                let deserializeGraphQLRequest () = task {
-                    try
-                        let! deserialized = JsonSerializer.DeserializeAsync<GraphQLRequest> (ctx.Request.Body, serializerOptions)
-                        return Ok deserialized
-                    with :? GraphQLException as ex ->
-                        logger.LogError (``exception`` = ex, message = "Error while deserializing request.")
-                        return Result.Error [ $"%s{ex.Message}\n%s{ex.ToString ()}" ]
-                }
 
-                let applyPlanExecutionResult (result : GQLExecutionResult) = task {
-                    let gqlResponse =
-                        match result.Content with
-                        | Direct (data, errs) -> GQLResponse.Direct (result.DocumentId, data, errs)
-                        | RequestError (problemDetailsList) -> GQLResponse.RequestError (result.DocumentId, problemDetailsList)
-                        | _ ->
-                            GQLResponse.RequestError (
-                                result.DocumentId,
-                                [
-                                    GQLProblemDetails.Create ("subscriptions are not supported here (use the websocket endpoint instead).")
-                                ]
+    let rec private moduleType = getModuleType <@ moduleType @>
+
+    let ofTaskIResult ctx (taskRes: Task<IResult>) : HttpFuncResult = task {
+        let! res = taskRes
+        do! res.ExecuteAsync(ctx)
+        return Some ctx
+    }
+
+    let ofTaskIResult2 ctx (taskRes: Task<Result<IResult, IResult>>) : HttpFuncResult =
+        taskRes
+        |> TaskResult.defaultWith id
+        |> ofTaskIResult ctx
+
+    /// Set CORS to allow external servers (React samples) to call this API
+    let setCorsHeaders : HttpHandler =
+        setHttpHeader "Access-Control-Allow-Origin" "*"
+        >=> setHttpHeader "Access-Control-Allow-Headers" "content-type"
+
+    let private handleGraphQL<'Root> (next : HttpFunc) (ctx : HttpContext) =
+        let sp = ctx.RequestServices
+
+        let logger = sp.CreateLogger moduleType
+
+        let options = sp.GetRequiredService<GraphQLOptions<'Root>>()
+
+        let toResponse { DocumentId = documentId; Content = content; Metadata = metadata } =
+
+            let serializeIdented value =
+                let jsonSerializerOptions = options.GetSerializerOptionsIdented()
+                JsonSerializer.Serialize(value, jsonSerializerOptions)
+
+            match content with
+            | RequestError errs ->
+                logger.LogInformation(
+                    $"Produced request error GraphQL response with documentId = '{{documentId}}' and metadata:{Environment.NewLine}{{metadata}}",
+                    documentId,
+                    metadata
+                )
+
+                GQLResponse.RequestError(documentId, errs)
+            | Direct(data, errs) ->
+                logger.LogInformation(
```

Comment:
Again, I think `Information` is too verbose here. This is a successful response, so I think `Debug` would be a good choice.